### PR TITLE
[Falaise] propose new event generator : Rh106

### PR DESCRIPTION
**Pull request by lemiere**
Analysis group asked for a new event generator : Rh106.
I added a more_backgroud file in primary_event/1.3 which aggregate tag 1.2.

This is a fix for issue  #97

### CODE REVIEW

File: resources/config/snemo/demonstrator/simulation/geant4_control/2.3/README.rst [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+   :depth: 3
+..
+
+Presentation
+============
+
+This directory  contains the files  needed to run Geant4 simulation
+for the SuperNEMO demonstrator simulation.
+
+* Version is : ``2.2``
```
Comment (drbenmorgan) [inline resources/config/snemo/demonstrator/simulation/geant4_control/2.3/README.rst]: Minor point - this doesn't match the tag of `2.3`

File: resources/config/snemo/demonstrator/simulation/primary_events/1.4/README.rst [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+..
+
+Presentation
+============
+
+This directory contains  the files needed to  describe several primary
+event generators as input for SuperNEMO simulation.
+
+  * Setup label is : ``snemo::demonstrator::simulation::primary_events``
+  * Version is : ``1.3``
```
Comment (drbenmorgan) [inline resources/config/snemo/demonstrator/simulation/primary_events/1.4/README.rst]: Minor point again: this doesn't match the tag in the path of `1.4`

File: resources/config/snemo/demonstrator/simulation/geant4_control/2.3/README.rst [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+Browse and edit primary event generation variant parameters and options
+===============================================================================
+
+1. From the build directory, generate documentation about variants:
+
+.. code:: sh
+
+   brewsh> bxvariant_inspector \
+          --datatools::resource-path="falaise@$(pwd)/BuildProducts/share/Falaise-3.0.0/resources" \
+          --variant-config "@falaise:config/snemo/demonstrator/simulation/geant4_control/2.2/variants/repository.conf" \
```
Comment (drbenmorgan) [inline resources/config/snemo/demonstrator/simulation/geant4_control/2.3/README.rst]: The versions/tags here don't match current ones (3.2.0 for Falaise, 2.3 for this tag).

File: resources/config/snemo/demonstrator/simulation/geant4_control/2.3/README.rst [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+..
+
+2. From the build directory, browse/edit the primary event generation variant and generate a variant profile::
+
+.. code:: sh
+
+   brewsh> bxvariant_inspector \
+	  --datatools::resource-path="falaise@$(pwd)/BuildProducts/share/Falaise-3.0.0/resources" \
+	  --variant-config "@falaise:config/snemo/demonstrator/simulation/geant4_control/2.2/variants/repository.conf" \
+          --variant-gui \
```
Comment (drbenmorgan) [inline resources/config/snemo/demonstrator/simulation/geant4_control/2.3/README.rst]: As above, non-current versions/tags

File: resources/config/snemo/demonstrator/simulation/geant4_control/2.3/README.rst [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+Run from  the Falaise build directory.  Here we **don't** use  the variant
+support but manually set vertex and particle generators associated to
+the default geometry:
+
+.. raw:: sh
+
+   brewsh> LD_LIBRARY_PATH="$(pwd)/BuildProducts/lib:${LD_LIBRARY_PATH}" \
+     bxg4_production \
+     --datatools::logging "warning" \
+     --datatools::resource-path "falaise@$(pwd)/BuildProducts/share/Falaise-3.0.0/resources" \
```
Comment (drbenmorgan) [inline resources/config/snemo/demonstrator/simulation/geant4_control/2.3/README.rst]: Old version

File: resources/config/snemo/demonstrator/simulation/geant4_control/2.3/README.rst [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+     --logging-priority "debug" \
+     --load-dll Falaise \
+     --config @falaise:config/snemo/demonstrator/simulation/geant4_control/2.2/manager.conf \
```
Comment (drbenmorgan) [inline resources/config/snemo/demonstrator/simulation/geant4_control/2.3/README.rst]: Not this tag (2.4), this carries on through the file...

Probably better to put the versions/tags as "placeholders" like `<FalaiseVersion>`, `<Tag>`, to minimise copy pasting work.

File: resources/config/snemo/demonstrator/simulation/primary_events/1.4/README.rst [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+Browse and edit primary event generation variant parameters and options
+===============================================================================
+
+1. From the build directory, generate documentation about variants:
+
+.. code:: sh
+
+   $ bxvariant_inspector \
+          --datatools::resource-path="falaise@$(pwd)/BuildProducts/share/Falaise-3.1.1/resources" \
+          --variant-config "@falaise:config/snemo/demonstrator/simulation/primary_events/1.3/variants/repository.conf" \
```
Comment (drbenmorgan) [inline resources/config/snemo/demonstrator/simulation/primary_events/1.4/README.rst]: As above, maybe change explicit versions to placeholders so changes can be confined to single place (e.g the Version in the Presentation section above.

**Comment by lemiere** [comment]
The versioning mismatches  --> corrected
The resources/_BuildResourceList.sh script should be run --> done
Ddo you want to make urn:snemo:demonstrator:simulation:2.3 --> no

a simple run of flsimulate with the new setup and generator should be added as a test. --> not necessary. But I will ask to Analysis Group (@emchauve ) to check the generator.

**Comment by drbenmorgan** [comment]
Thanks @lemiere! 

On the conflict it's now reporting, your `feature_add_rh106` branch needs rebasing against this repo's develop branch. Much the same procedure as described in this comment: https://github.com/SuperNEMO-DBD/Falaise/pull/100#issuecomment-386093072 can be applied. The rebase will complain about the conflict, but just rerun `_BuildResourceList.sh` and `git add` the regenerated `_resources.cmake` followed by `git rebase --continue` , no need to edit it by hand in this case!

> Ddo you want to make urn:snemo:demonstrator:simulation:2.3 --> no

O.k., let's review this when the next release is planned.

> a simple run of flsimulate with the new setup and generator should be added as a test. --> not necessary. But I will ask to Analysis Group (@emchauve ) to check the generator.

Right, I wasn't thinking of a validation test, just a simple 2-10 event run of `flsimulate` to confirm the new setup and generator. That's just to confirm the setup is runnable, with no conflicts, now and also in the future when additional setups are implemented. We have seen issues with existing/new config files (e.g. #21), so this is just a sanity check, nothing more, but is required!

**Comment by drbenmorgan** [comment]
@lemiere, just one error in the test, which looks to be a simple case of the 424045f commit not having the `.conf` file present (probably just a missing `git add`?).

**Comment by lemiere** [comment]
My bad!
Just done and file is now available

**Comment by drbenmorgan** [comment]
@lemiere, just one more typo issue I think:

https://github.com/lemiere/Falaise/blob/424045fed307748b05cb6aeecaeefc11d049076b/source/flsimulate/tests/flsimulate-issue97.profile#L20

**Comment by lemiere** [comment]
Aaaarg!
Done

**Pull request merged**